Gems/PhysX/Code/Source/EditorStaticRigidBodyComponent.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Serialization/EditContext.h>
+
+#include <Source/StaticRigidBodyComponent.h>
+#include <Source/EditorStaticRigidBodyComponent.h>
+
+namespace PhysX
+{
+    void EditorStaticRigidBodyComponent::Reflect(AZ::ReflectContext* context)
+    {
+        auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<EditorStaticRigidBodyComponent, AzToolsFramework::Components::EditorComponentBase>()
+                ->Version(1);
+
+            AZ::EditContext* editContext = serializeContext->GetEditContext();
+            if (editContext)
+            {
+                editContext
+                    ->Class<EditorStaticRigidBodyComponent>(
+                        "PhysX Static Rigid Body", "The entity behaves as a static rigid object in PhysX.")
```

Comment:
Do we want to add `non-movable` to this description as well?

Reply:
Good call, I'll add it, and I'll add `movable` to the description of dynamic rigid body too.